spring-ai-docs/src/main/antora/modules/ROOT/pages/api/vectordbs/elasticsearch.adoc

```diff
@@ -0,0 +1,138 @@
+
+[source,xml]
+----
+<dependency>
+    <groupId>org.springframework.ai</groupId>
+    <artifactId>spring-ai-elasticsearch-store-spring-boot-starter</artifactId>
+</dependency>
+----
+
+or to your Gradle `build.gradle` build file.
+
+[source,groovy]
+----
+dependencies {
+    implementation 'org.springframework.ai:spring-ai-elasticsearch-store-spring-boot-starter'
+}
+----
+
+The Vector Store, also requires an `EmbeddingClient` instance to calculate embeddings for the documents.
+You can pick one of the available xref:api/embeddings.adoc#available-implementations[EmbeddingClient Implementations].
+
+For example to use the xref:api/embeddings/openai-embeddings.adoc[OpenAI EmbeddingClient] add the following dependency to your project:
+
+[source,xml]
+----
+<dependency>
+	<groupId>org.springframework.ai</groupId>
+	<artifactId>spring-ai-openai-spring-boot-starter</artifactId>
+</dependency>
+----
+
+or to your Gradle `build.gradle` build file.
+
+[source,groovy]
+----
+dependencies {
+    implementation 'org.springframework.ai:spring-ai-openai-spring-boot-starter'
+}
+----
+
+TIP: Refer to the xref:getting-started.adoc#dependency-management[Dependency Management] section to add the Spring AI BOM to your build file.
+Refer to the xref:getting-started.adoc#repositories[Repositories] section to add Milestone and/or Snapshot Repositories to your build file.
+
+To connect to and configure Elasticsearch, you need to provide access details for your instance.
+A simple configuration can either be provided via Spring Boot's `application.yml`
+
+[source,yaml]
+----
+spring:
+  elasticsearch:
+    uris: <elasticsearch instance URIs>
+    username: <elasticsearch username>
+    password: <elasticsearch password>
+  ai:
+    vectorstore:
+      elasticsearch:
+        index-name: spring-ai-index
+----
+
+TIP: Check the list of <<elasticsearchvector-properties, configuration parameters>> to learn about the default values and configuration options.
+
+In your main code, create some documents:
+
+[source,java]
+----
+List<Document> documents = List.of(
+	new Document("Spring AI rocks!! Spring AI rocks!! Spring AI rocks!! Spring AI rocks!! Spring AI rocks!!", Map.of("meta1", "meta1")),
+	new Document("The World is Big and Salvation Lurks Around the Corner"),
+	new Document("You walk forward facing the past and you turn back toward the future.", Map.of("meta2", "meta2")));
+----
+
+Add the documents to Elasticsearch:
+
+[source,java]
+----
+vectorStore.add(List.of(document));
+----
+
+And finally, retrieve documents similar to a query:
+
+[source,java]
+----
+List<Document> results = vectorStore.similaritySearch(SearchRequest.query("Spring").withTopK(5));
+----
+
+If all goes well, you should retrieve the document containing the text "Spring AI rocks!!".
+
+[[elasticsearchvector-properties]]
+== ElasticsearchVectorStore Properties
+
+You can use the following properties in your Spring Boot configuration to customize the Elasticsearch Vector Store.
```

Comment:
I think we can split this section in `Configuration`, `Auto-Configuration` and `Metadata filtering` in order to be aligned with [Neo4j docs](https://docs.spring.io/spring-ai/reference/api/vectordbs/neo4j.html)